zerver/tests/test_invite.py

```diff
@@ -3123,6 +3483,98 @@ def test_create_multiuse_link_with_specified_user_groups_api_call(self) -> None:
+            name=SystemGroups.MEMBERS, realm=realm, is_system_group=True
+        )
+        nobody_group = NamedUserGroup.objects.get(
+            name=SystemGroups.NOBODY, realm=realm, is_system_group=True
+        )
+
+        do_change_realm_permission_group_setting(
+            realm, "can_add_subscribers_group", admins_group, acting_user=None
+        )
+
+        self.login("hamlet")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{multiuse_invite_1.id}",
+            {
+                "stream_ids": orjson.dumps(stream_ids).decode(),
+                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_error(result, "Insufficient permission")
+
+        self.login("iago")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{multiuse_invite_1.id}",
+            {
+                "stream_ids": orjson.dumps(stream_ids).decode(),
+                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_success(result)
+
+        do_change_realm_permission_group_setting(
+            realm, "can_add_subscribers_group", nobody_group, acting_user=None
+        )
+
+        self.login("iago")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{multiuse_invite_1.id}",
+            {
+                "stream_ids": orjson.dumps(stream_ids).decode(),
+                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_success(result)
+
+        do_change_stream_group_based_setting(
+            streams[0],
+            "can_add_subscribers_group",
+            members_group,
+            acting_user=self.example_user("iago"),
+        )
+
+        self.login("hamlet")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{multiuse_invite_1.id}",
+            {
+                "stream_ids": orjson.dumps([streams[0].id]).decode(),
+                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_error(result, "Insufficient permission")
+
+        result = self.client_patch(
+            f"/json/invites/multiuse/{multiuse_invite_1.id}",
+            {
+                "stream_ids": orjson.dumps(stream_ids).decode(),
+                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_error(result, "Insufficient permission")
```

Comment:
This should also include a test to check successful invite by having permission through stream-level setting.
You can set stream level setting for one stream, say stream 1 to a anonymous group containing hamlet, just so that we set it to a group with limited users than members group. Then set `stream_ids` field to include both streams that should fail and then set stream_ids field to only containing ID of stream 1 which should pass for hamlet but not for cordelia.